embulk-input-s3/src/main/java/org/embulk/input/s3/AbstractS3FileInputPlugin.java

```diff
@@ -241,6 +252,13 @@ private FileList listFiles(PluginTask task)
         }
     }
 
+    private void addS3DirectObject(FileList.Builder builder, AmazonS3 client, String bucket, String objectKey)
+    {
+        GetObjectMetadataRequest objectMetadataRequest = new GetObjectMetadataRequest(bucket, objectKey);
+        ObjectMetadata objectMetadata = client.getObjectMetadata(objectMetadataRequest);
```

Comment:
Since this is still a network call, should this be wrapped around a retry loop just like `listS3FilesByPrefix` does with `AmazonS3#listObjects`?

Reply:
Yeah, I though about it too, but the retry solution is being refactored completely, therefore it'd be better to do it right after retry successfully refactored.

Reply:
Ok, then fine

Reply:
`getObjectMetadata()` must be retried with `S3FileInputUtils.executeWithRetry`.

Reply:
Retry will be applied after https://github.com/embulk/embulk-input-s3/pull/72 is merged.